src/useGoogleAnalytics.ts

```diff
@@ -3,19 +3,31 @@ import GoogleAnalytics from 'react-ga4';
 import { useLocation } from 'react-router-dom';
 import { useOneTrust } from './useOneTrust';
 
-export const GoogleAnalyticsConsentValue = {
-    Denied: "denied",
-    Granted: "granted"
+export enum GoogleAnalyticsConsentValue {
+    Denied = "denied",
+    Granted = "granted"
 };
 
-export const useGoogleAnalytics = ({ googleAnalyticsId, debug, nonce, oneTrustScriptDomain }) => {
-    let location = useLocation();
-    const [initializeOneTrust] = useOneTrust({oneTrustScriptDomain, nonce});
+interface UseGoogleAnalyticsParams {
+    googleAnalyticsId?: string;
+    debug?: boolean;
+    nonce?: string;
+    oneTrustScriptDomain?: string;
+};
+
+export const useGoogleAnalytics = ({
+    googleAnalyticsId,
+    debug,
+    nonce,
+    oneTrustScriptDomain
+}: UseGoogleAnalyticsParams) => {
+    const location = useLocation();
+    const [initializeOneTrust] = useOneTrust({ oneTrustScriptDomain, nonce });
 
     const [initialized, setInitialized] = useState(false);
```

Comment:
isn't the type should be boolean?

Reply:
I wasn't sure if it's necessary to define type for initialized, since typescript would infer the boolean type based on the initial state being false?